File: cpp/test/PlayerTest.cpp

#include <string>
#include <vector>

#include "gtest/gtest.h"
#include "../model/Player.h"
#include "../model/Damage.h"
#include "../model/Inventory.h"
#include "../model/Stats.h"
#include "../model/SimpleEnemy.h"
#include "../model/BasicItem.h"
#include "../model/BasicBuff.h"
#include "../model/SimpleArmor.h"

using namespace std;

// Helpful test data which corresponds to items listed in the top level README file
BasicItem shield = BasicItem("round shield", 0, 1.4f);
BasicItem sword = BasicItem("Flashy sword of danger", 10, 1.0f);
BasicItem excalibur = BasicItem("Excalibur", 20, 1.5f);
BasicItem helmet = BasicItem("helmet of swiftness", 0, 1.2f);
BasicItem boots = BasicItem("ten league boots", 0, 0.1f);
BasicItem breastplate = BasicItem("breastplate of steel", 0, 1.4f);
std::vector<Buff *> buffs = {new BasicBuff(1.0f, 1.0f)};
SimpleArmor armor = SimpleArmor(5);
SimpleEnemy enemy = SimpleEnemy(&armor, buffs);


TEST(PlayerTest, DamageCalculations) {
    //TODO: finish this test case
    auto inventory = new Inventory(/*todo*/);
    auto stats = new Stats(0);
    auto target = new SimpleEnemy(/*todo*/, /*todo*/);
    auto player =  new Player(*inventory, *stats);
    
    Damage *damage = player->calculateDamage(*target);

    ASSERT_EQ(10, damage->getAmount());
}

File: misc/java/src/test/java/codingdojo/PlayerTest.java
package codingdojo;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.Mockito.mock;

public class PlayerTest {

    /*
     * Helpful test data which corresponds to items listed in the top level README file
     */
    public static class TestData
    {
        public static BasicItem shield = new BasicItem("round shield", 0, 1.4f);
        public static BasicItem sword = new BasicItem("Flashy sword of danger", 10, 1.0f);
        public static BasicItem excalibur = new BasicItem("Excalibur", 20, 1.5f);
        public static BasicItem helmet = new BasicItem("helmet of swiftness", 0, 1.2f);
        public static BasicItem boots = new BasicItem("ten league boots", 0, 0.1f);
        public static BasicItem breastplate = new BasicItem("breastplate of steel", 0, 1.4f);
        public static SimpleEnemy enemy = new SimpleEnemy(
                new SimpleArmor(5),
                List.of(new BasicBuff(1.0f, 1.0f))
        );
    }

    // choose this one if you are familiar with mocks
    @Disabled("Test is not finished yet")
    @Test
    void damageCalculationsWithMocks() {
        Inventory inventory = mock(Inventory.class);
        Stats stats = mock(Stats.class);
        SimpleEnemy target = mock(SimpleEnemy.class);

        Damage damage = new Player(inventory, stats).calculateDamage(target);
        assertEquals(10, damage.getAmount());
    }

    @Disabled("Test is not finished yet")
    @Test
    void damageCalculations() {
        Inventory inventory = new Inventory(null);
        Stats stats = new Stats(0);
        Player player = new Player(inventory, stats);

        Damage damage = player.calculateDamage(null);

        assertEquals(9, damage.getAmount());
    }
}
